python/cudf/cudf/pandas/module_accelerator.py

```diff
@@ -503,20 +506,19 @@ def disabled(self):
         -------
         Context manager for disabling things
         """
-        with self._use_fast_lib_lock:
-            # Have to hold the lock to modify this variable since
-            # another thread might be reading it.
-            # Modification has to happen with the lock held for the
-            # duration, so if someone else has modified things, then
-            # we block trying to acquire the lock (hence it is safe to
-            # release the lock after modifying this value)
-            saved = self._use_fast_lib
-            self._use_fast_lib = False
+        self._use_fast_lib_bkp[threading.get_ident()].append(
+            # The first time that a thread is disabled, unless it was the main
+            # thread it won't have an entry in the dictionary, which implicitly
+            # means it is running in fast mode.
+            self._use_fast_lib.get(threading.get_ident(), True)
+        )
+        self._use_fast_lib[threading.get_ident()] = False
         try:
             yield
         finally:
-            with self._use_fast_lib_lock:
-                self._use_fast_lib = saved
+            self._use_fast_lib[threading.get_ident()] = self._use_fast_lib_bkp[
+                threading.get_ident()
+            ].pop()
```

Comment:
A comment here could be useful.

Reply:
Could you suggest what you'd want in the comment? I'm not sure what I would write that wouldn't just be describing the line of code.

Reply:
I think my question here will help: https://github.com/rapidsai/cudf/pull/17955#discussion_r1947273113

Reply:
Does this restore the state of the thread? I think popping here get the last stored value.

Reply:
We resolved this in the other thread.